spec/presenters/iiif_print/iiif_manifest_presenter_factory_behavior_spec.rb

```diff
@@ -0,0 +1,51 @@
+require "spec_helper"
+
+RSpec.describe IiifPrint::IiifManifestPresenterBehavior do
+  let(:parent_fs_attributes) do
+    { "id" => "parent_fs123",
+      "title_tesim" => ["My Parent FileSet"],
+      "has_model_ssim" => ["FileSet"] }
+  end
+  let(:child_work_attributes) do
+    { "id" => "child_work123",
+      "title_tesim" => ["My Child Image"],
+      "has_model_ssim" => ["Image"],
+      "file_set_ids_ssim" => ["child_image_fs123"] }
+  end
+  let(:child_fs_attributes) do
+    { "id" => "child_fs123",
+    "title_tesim" => ["My Child FileSet"],
+    "has_model_ssim" => ["FileSet"] }
+  end
+  let(:parent_fs_solr_doc) { SolrDocument.new(parent_fs_attributes) }
+  let(:child_work_solr_doc) { SolrDocument.new(child_work_attributes) }
+  let(:child_fs_solr_doc) { SolrDocument.new(child_fs_attributes) }
+  let(:ids) { [parent_fs_solr_doc.id, child_work_solr_doc.id] }
+  let(:presenter_class) { Hyrax::IiifManifestPresenter }
+
+  subject(:presenter_factory) do
+    Hyrax::IiifManifestPresenter::Factory.new(
+      ids: ids,
+      presenter_class: presenter_class,
+      presenter_args: []
+    )
+  end
+
+  class MyWork; end
```

Comment:
Talk to me about this?  Not a blocker just an odd thing to see.

Reply:
I was getting a uninitialized constant when i use something like `Image` so i just decided to make a custom class instead, would a `class Image; end` be better in this case?

Reply:
There is a better mechanism to create an ephemeral class (and tidy up):

```ruby
around do |spec|
  class MyWork; end
  spec.run
  Object.send(:remove_const, :MyWork) 
end
```

The advantage of this setup is that the class only lives for the single test.  Make sure that `around` block is scoped to the tests that need the constant.

What can happen is that you can end up with multiple declarations of the class (technically re-openings of the class); which can be fine until one declaration says `class MyWork < ActiveFedora::Base; end` and then you have class mismatches.

Alternatively adding that class to the `spec_helper.rb` makes it available and only loaded/resolved once.